DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/SetDelegatePasswordControllerTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.TrackingSystem.Delegates
+{
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.DataServices.UserDataService;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates;
+    using DigitalLearningSolutions.Web.Tests.ControllerHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.SetDelegatePassword;
+    using FakeItEasy;
+    using FluentAssertions;
+    using FluentAssertions.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Primitives;
+    using NUnit.Framework;
+
+    public class SetDelegatePasswordControllerTests
+    {
+        private const string Password = "password";
+        private const int DelegateId = 2;
+
+        private IPasswordService passwordService = null!;
+        private HttpRequest request = null!;
+        private SetDelegatePasswordController setDelegatePasswordController = null!;
+        private IUserDataService userDataService = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            request = A.Fake<HttpRequest>();
+            userDataService = A.Fake<IUserDataService>();
+            passwordService = A.Fake<IPasswordService>();
+            setDelegatePasswordController = new SetDelegatePasswordController(passwordService, userDataService)
+                .WithMockHttpRequestHttpContext(request)
+                .WithMockServices()
+                .WithMockTempData()

```

Comment:
How come we need mock tempdata?

Reply:
It's needed in combination with the mock service, otherwise you get errors:
System.InvalidCastException : Unable to cast object of type 'Castle.Proxies.ObjectProxy_5' to type 'Microsoft.AspNetCore.Mvc.ViewFeatures.ITempDataDictionaryFactory'.
Irrelevant now as we don't need the mock http request any more, so they all go away.